tests/dist/python/run_dist_objects.py

```diff
@@ -0,0 +1,77 @@
+import dgl
+import torch
+import os
+import numpy as np
+import dgl.backend as F
+from dgl.distributed import load_partition_book
+import time
+
+mode = os.environ.get('DIST_DGL_TEST_MODE', "")
+graph_name = os.environ.get('DIST_DGL_TEST_GRAPH_NAME', 'random_test_graph')
+num_part = int(os.environ.get('DIST_DGL_TEST_NUM_PART', '2'))
+num_client_per_machine = int(os.environ.get('DIST_DGL_TEST_NUM_CLIENT', '2'))
+shared_workspace = os.environ.get('DIST_DGL_TEST_WORKSPACE')
+part_id = int(os.environ.get('DIST_DGL_TEST_PART_ID'))
+net_type = os.environ.get('DIST_DGL_TEST_NET_TYPE')
+
+os.environ['DGL_DIST_MODE'] = 'distributed'
+
+def zeros_init(shape, dtype):
+    return F.zeros(shape, dtype=dtype, ctx=F.cpu())
+
+
+def run_server(graph_name, server_id, server_count, num_clients, shared_mem, keep_alive=False):
+    g = dgl.distributed.DistGraphServer(server_id, shared_workspace + "ip_config.txt",
+                        server_count, num_clients,
+                        shared_workspace + 'dist_graph/{}.json'.format(graph_name),
+                        disable_shared_mem=not shared_mem,
+                        graph_format=['csc', 'coo'], keep_alive=keep_alive,
+                        net_type=net_type)
+    print('start server', server_id)
+    g.start()
+
+
+def dist_tensor_test_sanity(g, rank, name=None):
+    dist_ten = dgl.distributed.DistTensor((g.number_of_nodes(), 2),
+                                           F.int32,
+                                           init_func=zeros_init,
+                                           name=name)
+    if part_id == 0:
+        dist_ten[rank*3:(rank+1)*3] = F.ones((3, 2), dtype=F.int32, ctx=F.cpu()) * (rank+1)
+        dgl.distributed.client_barrier()
+    else:
+        dgl.distributed.client_barrier()
+        original_rank = rank // 2
+        assert F.allclose(dist_ten[original_rank*3:(original_rank+1)*3], F.ones((3, 2), dtype=F.int32, ctx=F.cpu()) * (original_rank+1))
+
+def test_dist_tensor(g, rank):
+    dist_tensor_test_sanity(g, rank)
+    dist_tensor_test_sanity(g, rank, name="SomeDistTensor")
+
+
+
+if mode == "server":
+    run_server(graph_name, server_id=part_id, server_count=1, num_clients=num_part*num_client_per_machine, shared_mem=False, keep_alive=False)
+elif mode == "client":
+    dgl.distributed.initialize(shared_workspace + "ip_config.txt")
+    global_rank = dgl.distributed.get_rank()
+    pid = os.getpid()
+    file_name = f"{pid}_rank_file_{global_rank}"
```

Comment:
why do we need to create these files? and these temp files should be removed when tests finish.